Sources/XiEditor/MarkerBar.swift

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class Marker {
+    let line: Int
+    let color: NSColor
+
+    init(_ line: Int, color: NSColor) {
+        self.line = line
+        self.color = color
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class MarkerBar: NSScroller {
+    var markers: [Marker] = []
+    var backgroundColor: NSColor = NSColor.red
+    let markerLayer = CAShapeLayer()
+    var overlayScrollerLayer: CALayer?
+    weak var markerDelegate: EditViewController?
+
+    override func viewWillDraw() {
+        super.viewWillDraw()
+
+        if (overlayScrollerLayer == nil) {
```

Comment:
This should make sure that we only add one observer, but I am not certain if this is the best way of doing this.

Reply:
Why viewWillDraw? It seems like we just want to add the observer once, when the view is created; viewWillDraw is going to get called _a lot_.

Reply:
I tried to set the observer in `init` but at that point `self.layer!.sublayers!.last` doesn't exist yet.

Reply:
Doesn't it get called when the user resizes the view?

Reply:
@mmatoszko `viewWillDraw` doesn't seem to get called when it is resized. I just added a `print` to the method to see how often it is executed and it only gets called once.